src/equations/acoustic_perturbation_2d.jl

```diff
@@ -0,0 +1,163 @@
+@doc raw"""
+    AcousticPerturbationEquations2D(v_mean, rho_mean, c_mean)
+
+!!! warning "Experimental code"
+    This system of equations is experimental and may change in any future release.
+
+Acoustic perturbation equations in two space dimensions with constant mean flow. `v_mean` is a tuple
+of size two holding the mean velocities in each direction. `rho_mean` and `c_mean` are scalars
+representing the mean density and mean speed of sound, respectively.
+
+The equations are based on the APE-4 system introduced in the following paper:
+
+R. Ewert, W. Schröder
+"Acoustic perturbation equations based on flow decomposition via source filtering",
+Journal of Computational Physics,
+Volume 188, Issue 2,
+2003,
+[DOI: 10.1016/S0021-9991(03)00168-2](https://doi.org/10.1016/S0021-9991(03)00168-2).
+"""
+struct AcousticPerturbationEquations2D{RealT<:Real} <: AbstractAcousticPerturbationEquations{2, 3}
+  v_mean::SVector{2, RealT}
+  rho_mean::RealT
+  c_mean::RealT
+end
+
+function AcousticPerturbationEquations2D(v_mean::NTuple{2,<:Real}, rho_mean::Real, c_mean::Real)
+  return AcousticPerturbationEquations2D(SVector(v_mean), rho_mean, c_mean)
+end
```

Comment:
The order `v_mean, rho_mean, c_mean` was a bit un-intuitive for me at first (since we have `rho, something_with_v, something_with_c` for the compressible Euler equations). I guess this order is standard in the field, @sloede? What about using a constructor with named variables?

Reply:
You have a valid point, but it is confusing either way: Here, we have `u'`, `v'`, `p'` as the conserved variables, thus it would be confusing either way. In general, I'd say that `v_mean` is the one quantity that needs to be changed most frequently, followed by `c_mean`. Thus, maybe as a middle way, we could order it `v_mean`, `c_mean`, `rho_mean` (the latter is often just `1`).

I also like the introduction of a ctor with named variables. Would you then make this an additional convenience ctor, @ranocha , or the only one?

Reply:
I would add it as a convenience constructor (and keep the standard inner constructor created automatically). You might also try to remove the outer constructor above since the tuple should be converted to an `SVector` automatically, but I'm not completely sure about this.

Reply:
Sounds good to me.

Reply:
I changed the order to `v_mean, c_mean, rho_mean` and added a constructor with named variables for convenience in 284d1ecce8a5ed2158fb767b1613812bf571f5db.

> You might also try to remove the outer constructor above since the tuple should be converted to an SVector automatically, but I'm not completely sure about this.

I tried it and got an error. It looks like it only works for non-parametric types:

```julia
using StaticArrays

struct SomeType
 vec::SVector{2, Float64}
end

struct ParametricType{RealT<:Real}
  vec::SVector{2, RealT}
end

julia> t1 = SomeType((1.0, 2.0))
SomeType([1.0, 2.0])

julia> t2 = ParametricType((1.0, 2.0))
ERROR: MethodError: no method matching ParametricType(::Tuple{Float64,Float64})
Closest candidates are:
  ParametricType(::SArray{Tuple{2},RealT,1,2}) where RealT<:Real at REPL[3]:2
Stacktrace:
 [1] top-level scope at REPL[5]:1
```

Reply:
In that case just leave the ctor in there and finish up the remaining issues.